client/post-editor/controller.js

```diff
@@ -179,6 +179,7 @@ async function maybeCalypsoifyGutenberg( context, next ) {
 		( isCalypsoifyGutenbergEnabled( state, siteId ) ||
 			isGutenlypsoEnabled( state, siteId ) ||
 			isEnabled( 'jetpack/gutenframe' ) ) &&
+		false !== isJetpackMinimumVersion( state, siteId, '7.3-alpha' ) &&
 		'gutenberg' === getSelectedEditor( state, siteId )
```

Comment:
at this point we're starting to pile up conditions and as a drive-by reader I'm finding it hard to keep straight why each one exists. can we consider adding explanatory comments indicating what purpose each condition fulfills?

I could even imagine it being helpful to rearrange this into a separate function, not for the runtime's sake, but for comprehension's sake

```js
const shouldRedirectToWPAdmin = ( state, siteId ) => {
	const notUsingNpmPackages = (
		isCalypsoifyGutenbergEnabled( state, siteId ) || // redirect to wp-admin with custom Calypso stylesheet
		isGutenlypsoEnabled( state, siteId ) || // not sure what this means
		isEnabled( 'jetpack/gutenframe' ) // we can load the editor in an iframe
	);

	const remoteSiteCanIframe = false !== isJetpackMinimumVersion( state, siteId, '7.3-alpha' );
	const userWantsGutenberg = 'Gutenberg' === getSelectedEditor( state, siteId );

	return ( notUsingNpmPackages && remoteSiteCanIframe && userWantsGutenberg );
}
```

honestly after digging in to this logic I'm confused and am not sure what we are trying to accomplish. it looks like we're in an historical pickle. a few lines up I see this function is named `maybeCalypsoifyGutenberg` but we might actually run this when _not_ Calypsoifying Gutenberg.

maybe it's time to back up a couple steps and reorganize and rename so that it's not unclear what the logical rules are and what the intended outcomes are?

Reply:
I split up the conditional in 90c7141, I hope that makes it a little more readable. Just to reiterate, the CalypsoifyGutenberg, Gutenlypso, and jetpack/gutenframe checks are temporary until the feature launched. Additionally, I changed the function name to something that felt a little more timeless. Very much open to improvement suggestions on all of that!